cachi2/core/models/sbom.py

```diff
@@ -395,27 +482,50 @@ class SPDXSbom(pydantic.BaseModel):
 
     creationInfo: SPDXCreationInfo
     packages: list[SPDXPackage] = []
+    files: list[SPDXFile] = []
     relationships: list[SPDXRelation] = []
 
+    def __hash__(self) -> int:
+        return hash(
+            hash(self.name + self.documentNamespace)
+            + hash(SPDXCreationInfo)
+            + sum(hash(p) for p in self.packages)
+            + sum(hash(f) for f in self.files)
+            + sum(hash(r) for r in self.relationships)
+        )
+
+    @classmethod
+    def from_file(cls, path: Path) -> "SPDXSbom":
+        """Consume a SPDX json directly from a file."""
+        return cls.model_validate_json(path.read_text())
+
     @staticmethod
     def deduplicate_spdx_packages(items: Iterable[SPDXPackage]) -> list[SPDXPackage]:
         """Deduplicate SPDX packages and merge external references.
 
         If package with same name and version is found multiple times in the list,
         merge external references of all the packages into one package.
         """
+        # NOTE: keeping this implementation mostly intact for historical reasons.
         unique_items = {}
         for item in items:
             purls = [
                 ref.referenceLocator for ref in item.externalRefs if ref.referenceType == "purl"
             ]
             if purls:
                 keys = [PackageURL.from_string(purl) for purl in purls if purl]
-                purl_tnv = [(p.type, p.name, p.version) for p in keys][0]
+                # name can exist in mutiple namespaces, e.g. rand exists in both math and
+                # crypto, and must be distinguished between.
+                p = keys[0]
+                package_name = p.name
+                if p.namespace is not None:
+                    package_name = p.namespace + "/" + p.name
+                purl_tnv = (p.type, package_name, p.version)
             else:
                 purl_tnv = ("", item.name, item.versionInfo or "")
 
             if purl_tnv not in unique_items:
+                # TODO: model_copy?
```

Comment:
If I were to deal with this module without prior knowledge or involvement (e.g. now) I wouldn't know what this TODO is about, can we be slightly more verbose?